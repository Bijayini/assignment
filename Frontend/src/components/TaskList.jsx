import React, { Component } from "react";

import Loader from "../ui-components/Loader";
import Error from "../pages/Error";

import EditTask from "./EditTask";
import DeleteTask from "./DeleteTask";

class TaskList extends Component{
  constructor(props){
    super(props);
  }

  componentDidMount(){
    this.props.fetchTasks();
  }

  componentDidUpdate(prevProps){
    const {shouldRefresh} = this.props;
    if(shouldRefresh){
      this.props.fetchTasks();
    }
  }

  render(){
    const {tasks,error, loading, deleteTask, editTask} = this.props;
    if (error) {
        return <Error message={error}/>
    }
    if (loading) {
      return <Loader/>;
    }

    return(
      <div className="main-content">
        <ul className="list-wrap wrap">
          {tasks.length ? tasks.map(task =>{
            return (<li key={task._id} className="list">
              <div className="list-header">
                <h3 className="heading">{task.title}</h3>
                <span className={`status ${task.status}`}>{task.status}</span>
              </div>
              <p className="description">{task.description}</p>
              <div className="hover-effect">
                <div className="button-container">
                  <EditTask editTask={editTask} task={task}/>
                  <DeleteTask handelClick={deleteTask} id={task._id} />
                </div>
              </div>
            </li>)
          }): <p>No task found</p>}
        </ul>
      </div>
    );
  }
}


export default TaskList;
