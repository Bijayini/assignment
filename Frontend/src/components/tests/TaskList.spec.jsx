import React from 'react';
import { shallow } from 'enzyme';
import TaskList from '../TaskList';


describe('TaskList', () => {
  let component;
  let props;

  beforeEach(()=>{
    props = {
      error:null,
      loading:false,
      fetchTasks:jest.fn(),
      tasks:[{
        description: 'some desc',
        status: 'some value',
        title: 'some title',
        _id:'23'
      }],
      deleteTask:jest.fn(),
      editTask:jest.fn()
    };
    component = shallow(<TaskList {...props}/>);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
    expect(props.fetchTasks).toHaveBeenCalled();
  });

  it('should render error component when error is not null', ()=>{
    props.error = 'some error';
    component = shallow(<TaskList {...props}/>);

    expect(component.find('Error').length).toBe(1);
  });

  it('should render loader component when loading is true', ()=>{
    props.loading = true;
    component = shallow(<TaskList {...props}/>);

    expect(component.find('Loader').length).toBe(1);
  });
});
