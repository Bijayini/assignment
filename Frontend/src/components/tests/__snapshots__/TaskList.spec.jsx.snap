// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList should match with snapshot 1`] = `
"<div className=\\"main-content\\">
  <ul className=\\"list-wrap wrap\\">
    <li className=\\"list\\">
      <div className=\\"list-header\\">
        <h3 className=\\"heading\\">
          some title
        </h3>
        <span className=\\"status some value\\">
          some value
        </span>
      </div>
      <p className=\\"description\\">
        some desc
      </p>
      <div className=\\"hover-effect\\">
        <div className=\\"button-container\\">
          <EditTask editTask={[Function: mockConstructor]} task={{...}} />
          <DeleteTask handelClick={[Function: mockConstructor]} id=\\"23\\" />
        </div>
      </div>
    </li>
  </ul>
</div>"
`;
